trinity/db/chain.py

```diff
@@ -0,0 +1,40 @@
+import asyncio
+import functools
+
+from multiprocessing.managers import (
+    BaseProxy,
+)
+
+
+def async_method(method_name):
+    async def method(self, *args, **kwargs):
+        loop = asyncio.get_event_loop()
+
+        return await loop.run_in_executor(
+            None,
+            functools.partial(self._callmethod, kwds=kwargs),
+            method_name,
+            args,
+        )
+    return method
+
+
+def sync_method(method_name):
```

Comment:
This is deserving of a metaclass which does these two things automatically but I'm going to set that aside as an issue someone else can come through and do the implementation for since this is totally functional as is.